testing/python/metafunc.py

```diff
@@ -434,6 +434,23 @@ def test_simple(x):
             "*uses no fixture 'y'*",
         ])
 
+    @pytest.mark.issue714
+    def test_parametrize_argument_not_in_indirect_list(self, testdir):
```

Comment:
I think we should have a test when `indirect` is just a string.

```python
@pytest.mark.issue714
def test_parametrize_argument_not_in_indirect_list(self, testdir):
    for decl in ("['x']", "'x'"):
        testdir.makepyfile("""
            import pytest
            @pytest.fixture(scope='function')
            def x(request):
                return request.param * 3
            @pytest.mark.parametrize('x, y', [('a', 'b')], indirect={decl}
            def test_simple(x):
                assert len(x) == 3
        """.format(decl=decl))
        result = testdir.runpytest("--collect-only")
        result.stdout.fnmatch_lines([
            "*uses no argument 'y'*",
        ])
```

It is a shame that we can't use parametrize here (we try to avoid using parametrize in this test module);